### ISSUE TITLE
 KDE prompts Remote Control Request muliple times and answer doesn't matter.

### ISSUE THREAD

**Issue by CupricReki** [feschber/lan-mouse#74]
I'm getting multiple Remote Control Request every time I launch lan-mouse from the CLI. 

## Log 
```
❯ lan-mouse
[2024-01-15T14:52:24Z INFO  lan_mouse::config] using config: "/home/cupric/.config/lan-mouse/config.toml"

(lan-mouse:1668930): Gtk-WARNING **: 09:52:24.175: Unknown key gtk-modules in /home/cupric/.config/gtk-4.0/settings.ini
[2024-01-15T14:52:24Z INFO  lan_mouse::config] using config: "/home/cupric/.config/lan-mouse/config.toml"
[2024-01-15T14:52:24Z INFO  lan_mouse] Press Ctrl+Alt+Shift+Super to release the mouse
[2024-01-15T14:52:24Z INFO  lan_mouse::consumer] wayland backend not available: the requested global was not found in the registry
[2024-01-15T14:52:24Z INFO  lan_mouse::producer] libei event producer not available: not implemented
[2024-01-15T14:52:24Z INFO  lan_mouse::producer] using layer-shell event producer

(lan-mouse:1668930): Adwaita-WARNING **: 09:52:24.261: Using GtkSettings:gtk-application-prefer-dark-theme with libadwaita is unsupported. Please use AdwStyleManager:color-scheme instead.
```
![KDE Screenshare](https://github.com/feschber/lan-mouse/assets/6509200/882e06df-058b-40be-895f-31baa2a798ae)
```
[2024-01-15T14:52:34Z INFO  lan_mouse::consumer] libei not available: Portal request failed: org.freedesktop.zbus.Error: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such method 'ConnectToEIS' in interface 'org.freedesktop.impl.portal.RemoteDesktop' at object path '/org/freedesktop/portal/desktop' (signature 'osa{sv}')
```
Remote Control Request prompt shows a second time.
```
[2024-01-15T14:52:40Z INFO  lan_mouse::consumer] using xdg-remote-desktop-portal event consumer
[2024-01-15T14:52:49Z INFO  lan_mouse] terminating service
[2024-01-15T14:52:49Z INFO  lan_mouse::server] terminating service
[2024-01-15T14:52:49Z INFO  lan_mouse::server] terminating gracefully...
```
## Notes
- Interestingly, it doesn't matter how if I accept or not, lan-mouse still works. 
- When launching via lan-mouse I don't always get the prompt, but I do get a notification. 
![KDE Screenshare popup](https://github.com/feschber/lan-mouse/assets/6509200/1dcbb2fe-d9c0-4b6d-b5d3-cc2f4898cde0)

**Comment by feschber** [comment]
The reason it shows up twice is that it first tries to connect to Libei, which is done through the same RemoteDesktopPortal interface as the actual remote desktop portal backend.

KDE does not implement Libei yet so that fails. Then the remote desktop portal requests access again.

It would be possible to work around this by simply requesting the remote desktop portal on KDE directly instead of Libei (I actually did this in earlier versions until I changed it to this fallback based system) but I'd rather leave it like this until KDE supports libei (I dont know how long that is going to take maybe I can look into it if I have the time).

As far as I have seen, KDE does not ask again once the permission has been granted once for a specific application so I dont think this should be a large issue but let me know your thoughts.

**Comment by CupricReki** [comment]
Bummer about Libei. But it looks like it might come to Plasma 6.1 https://invent.kde.org/plasma/xdg-desktop-portal-kde/-/issues/12.

**Comment by feschber** [comment]
> I'm not sure what triggers the permission prompt to be again after being accepted the first time. Perhaps the issue will go away now that that I'm installing/updating via the AUR.
> 
> Keeping this open for now.

Did you test this? I am not getting prompts after accepting it for the first time.

**Comment by feschber** [comment]
something must have changed because I see the prompt everytime now as well. Will see what I can do about that

**Comment by CupricReki** [comment]
Did you update to Plasma 6?

**Comment by feschber** [comment]
Yes but I can not say 100% if that was the point where it started. Will need to check

**Comment by svillar** [comment]
I've seen the same in GNOME FWIW

**Comment by feschber** [comment]
What version of GNOME are you using? Starting at 45, it should only ask once as libei is now supported.

I also changed the behaviour in 0.7 now, such that it asks again if the request is cancelled. So technically it now works as intended.

I could maybe look into merging the backends into one or reuse the same portal but idk if thats actually possible even.

**Comment by svillar** [comment]
> What version of GNOME are you using? Starting at 45, it should only ask once as libei is now supported.

Ah maybe that's the reason, I've 43 in Debian right now. Actually I couldn't make the software work at all, clients are detected but cannot share the keyboard/mouse among them. I've installed libei in all the computers but shows still as unsupported

**Comment by feschber** [comment]
Yeah libei alone won't help because gnome does not implement the input capture portal until version 45.

**Comment by svillar** [comment]
> Yeah libei alone won't help because gnome does not implement the input capture portal until version 45.

And should it work using any other fallback? I'm just trying to know whether this mix of GNOME 45 with 43 should work and then the problem is my configuration, or instead, I should wait till GNOME 45 in stable

**Comment by feschber** [comment]
> And should it work using any other fallback? 

Unfortunately not, no. Gnome does not support the layer-shell protocol.

**Comment by CupricReki** [comment]
Since updating to Plasma 6 I see two prompts every time we I start lan mouse

**Comment by hoergen** [comment]
I'm running Plasma 6.1.3 / Framework 6.4.0 and Wayland and still getting the request every time I start lan-mouse. I had to create "/.config/lan-mouse/config.toml" manually, because it wasn't created automatically.

When I start lan-mouse on two computers they can't find each other.

This is what I get, when I start lan-mouse in the terminal. There are a lot errors, but I don't know how to solve it. 
Maybe somebody else, already have an idea how to solve it?
```

./lan-mouse 
[2024-08-01T10:07:04Z INFO  lan_mouse::config] using config: "/home/USER/.config/lan-mouse/config.toml"
[2024-08-01T10:07:04Z INFO  lan_mouse] release bind: [KeyLeftCtrl, KeyLeftShift, KeyLeftMeta, KeyLeftAlt]
[2024-08-01T10:07:04Z INFO  lan_mouse::config] using config: "/home/USER/.config/lan-mouse/config.toml"
[2024-08-01T10:07:04Z INFO  lan_mouse] release bind: [KeyLeftCtrl, KeyLeftShift, KeyLeftMeta, KeyLeftAlt]
[2024-08-01T10:07:04Z INFO  lan_mouse] Press [KeyLeftCtrl, KeyLeftShift, KeyLeftMeta, KeyLeftAlt] to release the mouse
[2024-08-01T10:07:04Z INFO  lan_mouse::server] running service
[2024-08-01T10:07:04Z INFO  lan_mouse::server::emulation_task] creating input emulation...
[2024-08-01T10:07:04Z WARN  input_emulation] wlroots backend: `wayland protocol "wlr-virtual-pointer-unstable-v1" not supported: the requested global was not found in the registry`

(lan-mouse:18504): Gtk-WARNING **: 12:07:04.162: Unknown key gtk-modules in /home/USER/.config/gtk-4.0/settings.ini

(lan-mouse:18504): Gtk-WARNING **: 12:07:04.178: Theme directory places/symbolic of theme tuxedo-breeze-dark has no size field

[2024-08-01T10:07:04Z INFO  input_emulation::libei] requesting permission for input emulation

(lan-mouse:18504): Adwaita-WARNING **: 12:07:04.295: Using GtkSettings:gtk-application-prefer-dark-theme with libadwaita is unsupported. Please use AdwStyleManager:color-scheme instead.

(lan-mouse:18504): Gtk-WARNING **: 12:07:04.371: Theme directory places/symbolic of theme tuxedo-breeze-dark has no size field

[2024-08-01T10:07:07Z WARN  input_emulation] libei backend: `Portal request failed: org.freedesktop.zbus.Error: GDBus.Error:org.freedesktop.DBus.Error.Failed: Failed to connect to to EIS`
[2024-08-01T10:07:07Z INFO  input_emulation::xdg_desktop_portal] requesting permission for input emulation
[2024-08-01T10:07:08Z INFO  input_emulation] using emulation backend: xdg-desktop-portal
[2024-08-01T10:07:10Z WARN  input_capture] input-capture-portal input capture backend unavailable: error creating input-capture-portal backend: `xdg-desktop-portal: `Portal request didn't succeed: Other``
[2024-08-01T10:07:10Z INFO  input_capture] using capture backend: layer-shell
[2024-08-01T10:07:10Z INFO  lan_mouse::server::capture_task] []

```

**Comment by feschber** [comment]
Did you configure your clients? You need to add the corresponding device on both ends (device B on device A and device A on device B).

**Comment by gnespolino** [comment]
I'm also affected by this.
Not clear why, but 2 PC with same OS (Manjaro) and same updates, on one of them I can't enable input capture. So what happen when I configure clients on both, is that I can move from PC A to B, but I can't go back from B to A, B being the one where input capture cannot be enabled, mouse and keyboard connected to PC A.
This is what I get in the console when I click on "Reenable" and I allow input capture:

`
[2024-10-28T13:52:36Z INFO  lan_mouse::server] received capture enable request
[2024-10-28T13:52:44Z WARN  input_capture] input-capture-portal input capture backend unavailable: error creating input-capture-portal backend: `xdg-desktop-portal: `Portal request didn't succeed: Other``
[2024-10-28T13:52:44Z WARN  input_capture] layer-shell input capture backend unavailable: error creating layer-shell capture backend: `Could not find wayland compositor`
[2024-10-28T13:52:44Z WARN  input_capture] X11 input capture backend unavailable: error creating x11 capture backend: `X11 input capture is not yet implemented :(`
[2024-10-28T13:52:44Z WARN  lan_mouse::server::capture_task] input capture exited: error creating input-capture: `no backend available`
`

I also get the same output when launching lan-mouse on PC B.
When I launch lan-mouse on PC A, I don't get this lines in the console.

On both lan-mouse is installed through repositories
